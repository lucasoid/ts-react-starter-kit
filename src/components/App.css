body {
    background-color: #efefef;
}
.App {
    font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Helvetica, Arial, sans-serif, 'Apple Color Emoji',
        'Segoe UI Emoji', 'Segoe UI Symbol';
    font-size: 14px;
    width: 100%;
    max-width: 1080px;
    margin: 0 auto;
    color: #444;
}

.header {
    background-color: #fff;
    padding: 1em;
}

h1,
h2,
h3,
h4,
h5,
h6 {
    font-weight: 500;
}

a {
    color: teal;
    text-decoration: none;
}
