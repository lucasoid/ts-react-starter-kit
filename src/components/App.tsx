import * as React from 'react';
import { BrowserRouter as Router, Switch, Route } from 'react-router-dom';
import { MainMenu } from '~components/MainMenu';
import { ShoppingListMenu } from '~components/views/ShoppingListMenu';
import * as styles from './App.css';

interface AppProps {
    title: string;
}

export const App: React.FC<AppProps> = props => (
    <Router>
        <div className={styles.App}>
            <header className={styles.header}>
                <h1 className={styles.title}>{props.title}</h1>
            </header>
            <MainMenu />
            <Switch>
                <Route path="/settings">Settings controlled here.</Route>
                <Route path="/help">Help provided here.</Route>
                <Route path="/">
                    <ShoppingListMenu />
                </Route>
            </Switch>
        </div>
    </Router>
);
