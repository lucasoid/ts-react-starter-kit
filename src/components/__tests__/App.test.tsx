import * as React from 'react';
import { shallow } from 'enzyme';
import { App } from '../App';

describe('App', () => {
    it('renders the app', () => {
        let wrapper = shallow(<App title="Welcome" />);
        expect(wrapper).toMatchSnapshot();
    });
});
