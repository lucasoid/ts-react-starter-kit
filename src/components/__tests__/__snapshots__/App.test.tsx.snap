// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders the app 1`] = `
<BrowserRouter>
  <Theme>
    <div
      className="App"
    >
      <Header
        title="Welcome"
      />
      <ErrorBoundary>
        <Switch>
          <Route
            component={[Function]}
            path="/settings"
          />
          <Route
            component={[Function]}
            path="/about"
          />
          <Route
            component={[Function]}
            path="/lists/:listId"
          />
          <Route
            component={[Function]}
            exact={true}
            path="/"
          />
        </Switch>
      </ErrorBoundary>
    </div>
  </Theme>
</BrowserRouter>
`;
